app/src/main/java/com/poupa/vinylmusicplayer/adapter/SearchAdapter.java

```diff
@@ -81,46 +83,74 @@ public void onBindViewHolder(@NonNull final ViewHolder holder, int position) {
                 final Album album = (Album) dataSet.get(position);
                 holder.title.setText(album.getTitle());
                 holder.text.setText(MusicUtil.getAlbumInfoString(activity, album));
-                GlideApp.with(activity)
-                        .asDrawable()
-                        .load(VinylGlideExtension.getSongModel(album.safeGetFirstSong()))
-                        .apply(RequestOptions.bitmapTransform(new RoundedCorners(activity.getResources().getInteger(R.integer.rounded_corner))))
-                        .transition(VinylGlideExtension.getDefaultTransition())
-                        .songOptions(album.safeGetFirstSong())
-                        .into(holder.image);
+                if (PreferenceUtil.getInstance().useMaterialDesign()) {
+                    GlideApp.with(activity)
+                            .asDrawable()
+                            .load(VinylGlideExtension.getSongModel(album.safeGetFirstSong()))
+                            .apply(RequestOptions.bitmapTransform(new RoundedCorners(
```

Comment:
Having two branches (useMaterial vs not) makes code maintenance more tedious and error prone.

Is it possible to have a polymorphic `.apply(RequestOption...)` that hides this condition?

Reply:
Sure, I can do it for the roundedCorner
However, for the .circleCrop(), i am not sure